src/ConfigurationFactory.php

```diff
@@ -176,7 +178,13 @@ protected function createWithConfig(ContainerInterface $container, string $confi
         if ($config['middlewares'] !== []) {
             $middlewares = [];
             foreach ($config['middlewares'] as $middleware) {
-                $middlewares[] = is_string($middleware) ? $container->get($middleware) : $middleware;
+                if (is_string($middleware)) {
+                    $middlewares[] = $container->get($middleware);
+                } elseif ($middleware instanceof Middleware) {
```

Comment:
IMO not to be handled - should always be `string` in config, in my opinion :thinking:

Reply:
That wouldn't conform I think with other configuration values that are accepted as both a container id string, or as the actual instance (ex. https://github.com/Roave/psr-container-doctrine/blob/3.5.x/src/ConfigurationFactory.php#L116).

But, maybe that's a "feature" no one needs, I've made the change to accept only container id strings :+1:

Reply:
@samuelnogueira yeah, this library has been passed from hand to hand, so I'm trying to keep it slim/simple, heh :D 

The more we add, the more we have to support later on, and the harder it becomes to migrate, so I'm really trying to cut everything that isn't strictly vital.